pytorch_lightning/trainer/trainer.py

```diff
@@ -1152,7 +1155,8 @@ def _pre_training_routine(self):
         # register signals
         self.signal_connector.register_signal_handlers()
 
-        self.checkpoint_connector.resume_end()
+        if self.state.fn != TrainerFn.TUNING:
+            self.checkpoint_connector.resume_end()
```

Comment:
Given the objective of this PR, this now needs to move outside and not only be called for training. (_pre_training_routine is only called for training)

Reply:
that would actually create a problem. if someone do this
```py
Trainer(resume_from_checkpoint=...)
trainer.validate()
trainer.fit()  # here resume_from_checkpoint will be gone since validate will call resume_end and will destroy resume_from_checkpoint
```
this is a bit misalignment and should not work like this, but I think unify_checkpoint PR will clean this up in a better way.